### PR TITLE
assignment/scf-01-develope

### PR DESCRIPTION
**Pull request by meta-kritika-pandey4**

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Hitesh-Khatri-Metacube) [inline .classpath]: please delete this file from you commit

File: src/com/metacube/get2019/shop.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
+package com.metacube.get2019;
+
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: Single line spacing is enough here.

File: src/com/metacube/get2019/shop.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
+package com.metacube.get2019;
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: Add  Beginning comments for each file.

File: src/com/metacube/get2019/shop.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
+package com.metacube.get2019;
+
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Scanner;
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: add a line break after importing all packages.

File: src/com/metacube/get2019/shop.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
+package com.metacube.get2019;
+
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Scanner;
+class Item
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: Add class level comment before defining a class and define access modifier public/private/protected.

File: src/com/metacube/get2019/shop.java [hunk 6]
```diff
@@ -0,0 +1,250 @@
+package com.metacube.get2019;
+
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Scanner;
+class Item
+{
+int id;
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: provide access modifier while defining a global variable.

File: src/com/metacube/get2019/shop.java [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+double price;
+
+/*
+* This method is used to set values of 3 properties and to print them.
+* @param id This is first parameter to set id of item
+* @param itemName this is second parameter to set itemname
+* @param price this is third parameter to set price of item
+*/
+public Item(int id, String itemName, double price){
+this.id=id;
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: Follow indentation in your entire code.

File: src/com/metacube/get2019/shop.java [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+ * This class also has all the functions of a Shopping Cart
+ */
+class ShoppingCart
+{
+	//ArrayList are for storing the input
+	ArrayList<String> pro=new ArrayList<String>();
+	ArrayList<Integer> quant=new ArrayList<Integer>();
+	ArrayList<Integer> pid=new ArrayList<Integer>();
+	ArrayList<Double> price=new ArrayList<Double>();
+	Scanner ab=new Scanner(System.in);
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: You did not close your scanner after utilizing it, if you avoid this your code will not release the resource and may cause a performance issue.

Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: What is the reason here defining scanner as a global variable?

File: src/com/metacube/get2019/shop.java [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+			System.out.println("3.Remove from the cart");
+			System.out.println("4.Show items in the cart");
+			System.out.println("5.Make Bill"); 
+			System.out.println("6.Exit");
+			Scanner sc=new Scanner(System.in);
+			choice=sc.nextInt();
+			switch(choice)
+			{
+				case 1:
+
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: remove this whit space.

File: src/com/metacube/get2019/shop.java [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+			System.out.println("Choose the operation needed:");
+			System.out.println("1.Add in the cart");
+			System.out.println("2.Update the cart");
+			System.out.println("3.Remove from the cart");
+			System.out.println("4.Show items in the cart");
+			System.out.println("5.Make Bill"); 
+			System.out.println("6.Exit");
+			Scanner sc=new Scanner(System.in);
+			choice=sc.nextInt();
+			switch(choice)
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: What happens with the code if I provide -ve or non integer value?

File: src/com/metacube/get2019/shop.java [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+	ArrayList<Double> price=new ArrayList<Double>();
+	Scanner ab=new Scanner(System.in);
+	/*
+	 * addItem() adds items in the cart 
+	 * while taking the input from the customer
+	 */
+	void addItem(ArrayList<Item> itemList)
+	{
+		System.out.print("enter the item number = ");
+		Scanner ab=new Scanner(System.in);
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: You did not close your scanner after utilizing it, if you avoid this your code will not release the resource and may cause a performance issue

File: src/com/metacube/get2019/shop.java [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+		int itemId=ab.nextInt();
+		Iterator<Item> iterator=itemList.iterator();
+		while(iterator.hasNext())
+		{	
+			Item i=(Item)iterator.next();
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: naming convention should be proper.

File: src/com/metacube/get2019/shop.java [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+			if (i.getid()==itemId)
+			{
+				pro.add(i.getItemName());
+				System.out.print("Enter quantity for the item = ");
+				int itemQuantity=ab.nextInt();
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: What happens with the code if I provide -ve,String and decimal and out of rang values for example 2,147,483,650

File: src/com/metacube/get2019/shop.java [hunk 14]
```diff
@@ -0,0 +1,250 @@
...
+class Shop
+{
+	public static void main(String args[])
+	{
+		ArrayList<Item> itemList=new ArrayList<Item>();
+		ShoppingCart obj=new ShoppingCart();
+		int choice;
+		
+		System.out.println("The items available in the shop are:");
+		itemList.add(new Item(1,"Book",1000.00));
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: Print the table header as well to let the user know what are these details. 
Such as No, Item Name and Price.

File: src/com/metacube/get2019/shop.java [hunk 15]
```diff
@@ -0,0 +1,250 @@
...
+	ArrayList<Integer> pid=new ArrayList<Integer>();
+	ArrayList<Double> price=new ArrayList<Double>();
+	Scanner ab=new Scanner(System.in);
+	/*
+	 * addItem() adds items in the cart 
+	 * while taking the input from the customer
+	 */
+	void addItem(ArrayList<Item> itemList)
+	{
+		System.out.print("enter the item number = ");
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: First letter of a sentence should be in caps.

File: src/com/metacube/get2019/shop.java [hunk 16]
```diff
@@ -0,0 +1,250 @@
...
+	 */
+	void update()
+	{
+		if(pid.size()==0)
+		{
+			System.out.println("No item in the cart!! Please first enter some!!");
+		}
+		else
+		{
+			System.out.println("Enter the Sid of the product to be updated");
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: Do not confuse the user, while adding the item in the card while adding the item on line no 70  you mentioned enter the item number and you are asking end user to provide the SID, try to be stick on the same detail.

File: src/com/metacube/get2019/shop.java [hunk 17]
```diff
@@ -0,0 +1,250 @@
...
+/*
+ * This class has the attributes pro(Product),quant(Quantity), 
+ * pid(ProductID) and price(Cost) 
+ * This class also has all the functions of a Shopping Cart
+ */
+class ShoppingCart
+{
+	//ArrayList are for storing the input
+	ArrayList<String> pro=new ArrayList<String>();
+	ArrayList<Integer> quant=new ArrayList<Integer>();
```
Comment (Hitesh-Khatri-Metacube) [inline src/com/metacube/get2019/shop.java]: Isn't there a better way to deal with the cartItem instead of defining arrays globally?

**Review by Hitesh-Khatri-Metacube — CHANGES_REQUESTED** [review]
Hi Kritika, 

Provided my feedback, the one thing what I want to mention to make separate java file for Item , shop and shopping cart class to understand the code modularity and follow the separation of concern.

File: src/com/metacube/get2019/shop.java (hunk 2 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: done

File: src/com/metacube/get2019/shop.java (hunk 3 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: beginning comments added

File: src/com/metacube/get2019/shop.java (hunk 4 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: line is added

File: src/com/metacube/get2019/shop.java (hunk 5 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: access modifiers added

File: src/com/metacube/get2019/shop.java (hunk 6 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: done

File: src/com/metacube/get2019/shop.java (hunk 7 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: followed

File: src/com/metacube/get2019/shop.java (hunk 8 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: scanner is closed

Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: by mistake. mistake resolved

File: src/com/metacube/get2019/shop.java (hunk 9 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: removed whitespace

File: src/com/metacube/get2019/shop.java (hunk 10 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: exception handling taken care of

File: src/com/metacube/get2019/shop.java (hunk 11 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: resolved

File: src/com/metacube/get2019/shop.java (hunk 12 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: done

File: src/com/metacube/get2019/shop.java (hunk 13 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: exception handling taken care of

File: src/com/metacube/get2019/shop.java (hunk 14 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: printed

File: src/com/metacube/get2019/shop.java (hunk 15 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: done

File: src/com/metacube/get2019/shop.java (hunk 16 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: rectified

File: src/com/metacube/get2019/shop.java (hunk 17 above)
Comment (meta-kritika-pandey4) [reply src/com/metacube/get2019/shop.java]: added for the future extensibility

**Pull request merged**